source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,372 @@
+  addInfoToStream("propertyValue_circuitBreakerForceClosed", "true", cluster_info);
+  addStringToStream("propertyValue_executionIsolationStrategy", "SEMAPHORE", cluster_info);
+  addIntToStream("propertyValue_executionIsolationThreadTimeoutInMilliseconds", 0, cluster_info);
+  addInfoToStream("propertyValue_executionIsolationThreadInterruptOnTimeout", "false",
+                  cluster_info);
+  addIntToStream("propertyValue_executionIsolationSemaphoreMaxConcurrentRequests",
+                 max_concurrent_requests, cluster_info);
+  addIntToStream("propertyValue_fallbackIsolationSemaphoreMaxConcurrentRequests", 0, cluster_info);
+  addInfoToStream("propertyValue_requestCacheEnabled", "false", cluster_info);
+  addInfoToStream("propertyValue_requestLogEnabled", "true", cluster_info);
+  addIntToStream("reportingHosts", reporting_hosts, cluster_info);
+  addIntToStream("propertyValue_metricsRollingStatisticalWindowInMilliseconds",
+                 rolling_window_ms.count(), cluster_info);
+
+  ss << "data: {" << cluster_info.str() << "}" << std::endl << std::endl;
+}
+
+void HystrixSink::addHystrixThreadPool(absl::string_view cluster_name, uint64_t queue_size,
+                                       uint64_t reporting_hosts,
+                                       std::chrono::milliseconds rolling_window_ms,
+                                       std::stringstream& ss) {
+  std::stringstream cluster_info;
+
+  addIntToStream("currentPoolSize", 0, cluster_info);
+  addIntToStream("rollingMaxActiveThreads", 0, cluster_info);
+  addIntToStream("currentActiveCount", 0, cluster_info);
+  addIntToStream("currentCompletedTaskCount", 0, cluster_info);
+  addIntToStream("propertyValue_queueSizeRejectionThreshold", queue_size, cluster_info);
+  addStringToStream("type", "HystrixThreadPool", cluster_info);
+  addIntToStream("reportingHosts", reporting_hosts, cluster_info);
+  addIntToStream("propertyValue_metricsRollingStatisticalWindowInMilliseconds",
+                 rolling_window_ms.count(), cluster_info);
+  addStringToStream("name", cluster_name, cluster_info);
+  addIntToStream("currentLargestPoolSize", 0, cluster_info);
+  addIntToStream("currentCorePoolSize", 0, cluster_info);
+  addIntToStream("currentQueueSize", 0, cluster_info);
+  addIntToStream("currentTaskCount", 0, cluster_info);
+  addIntToStream("rollingCountThreadsExecuted", 0, cluster_info);
+  addIntToStream("currentMaximumPoolSize", 0, cluster_info);
+
+  ss << "data: {" << cluster_info.str() << "}" << std::endl << std::endl;
+}
+
+void HystrixSink::addClusterStatsToStream(ClusterStatsCache& cluster_stats_cache,
+                                          absl::string_view cluster_name,
+                                          uint64_t max_concurrent_requests,
+                                          uint64_t reporting_hosts,
+                                          std::chrono::milliseconds rolling_window_ms,
```

Comment:
nit: this one is my bad.  I think we can remove the `_ms` from this variable now that the type tells us that it's milliseconds.  Sorry about that!

Reply:
I left it to distinguish from rolling_window of RollingWindow type. Here and in some other places: https://github.com/trabetti/envoy/blob/d464f8c1f9de8eb21775ee1768006008543a21c6/source/extensions/stat_sinks/hystrix/hystrix.h#L57

Reply:
Okay, sounds good.  Not a huge deal to me either way.